### PR TITLE
[GEOS-9700] Cleaning up use of Eclipse EMF depdencies.

### PR DESCRIPTION
**Pull request by jnh5y**
Signed-off-by: Jim Hughes <jnh5y@ccri.com>

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [ ] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by jodygarnett** [comment]
@jnh5y I would like to accept this PR, but need you to go through the checklist (need a jira ticket for example).

File: pom.xml [hunk 1]
```diff
@@ -93,6 +93,7 @@
     <jts.version>1.17.0</jts.version>
     <jaiext.version>1.1.16</jaiext.version>
     <netcdf.version>4.6.15</netcdf.version>
+    <eclipse.emf.version>2.15.0</eclipse.emf.version>
```
Comment (jodygarnett) [inline pom.xml:96]: So this change does not actually change the version number, just adjust to reflect xsd dependency changing to group `org.eclipse.emf` eh?

Comment (jnh5y) [reply pom.xml:96]: The version number and groupId for org.eclipse.xsd both changed.

I wanted to pull out this version as a property to make it easier to coordinate between GT and GS.

**Comment by jodygarnett** [comment]
Update, this introduced some transitive dependencies into the build, excluding with:
```
    <dependency>
      <groupId>org.eclipse.emf</groupId>
      <artifactId>org.eclipse.xsd</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.eclipse.platform</groupId>
          <artifactId>org.eclipse.core.runtime</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

If this works I will remake the release

**Comment by jodygarnett** [comment]
Reported bug as https://bugs.eclipse.org/bugs/show_bug.cgi?id=565408

**Pull request merged**